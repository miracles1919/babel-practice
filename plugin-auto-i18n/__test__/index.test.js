const fs = require('fs');
const path = require('path');
const pluginTester = require('babel-plugin-tester').default;

const AutoI18nPlugin = require('../plugin/index');

const source = fs
  .readFileSync(path.join(__dirname, '../source.jsx'))
  .toString();

pluginTester({
  plugin: AutoI18nPlugin,
  pluginName: 'AutoI18nPlugin',
  pluginOptions: {
    output: '../locale',
  },
  babelOptions: {
    parserOpts: {
      sourceType: 'unambiguous',
      plugins: ['jsx'],
    },
  },
  tests: {
    demo1: {
      code: source,
      snapshot: true,
    },
  },
});
