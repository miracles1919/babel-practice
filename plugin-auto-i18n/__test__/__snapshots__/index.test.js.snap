// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoI18nPlugin demo1: demo1 1`] = `

import React from 'react';
import { Button, Space } from 'components';
import { DemoBlock } from 'demos';
import { SearchOutline } from 'icons';


export default () => {
  // 注释a
  const a = '带图标的按钮'
  const b = \`禁用状态\` // 注释b
  const abc = \`\${a}\${b}带图标的按钮\`

  return (
    <>
      <DemoBlock title='带图标的按钮'>
        <Button>
          <Space>
            <SearchOutline />
            <span>搜索</span>
          </Space>
        </Button>
      </DemoBlock>

      <DemoBlock title='禁用状态'>
        <Space wrap>
          <Button disabled>Disabled</Button>
          <Button disabled color=/*i18n-disable*/'primary'>
            Disabled
          </Button>
        </Space>
      </DemoBlock>

      <DemoBlock title={\`\${loadingTitle}状态\`}>
        <Space wrap>
          <Button loading color='primary' loadingText='正在加载'>
            Loading
          </Button>
          <Button loading>Loading</Button>
        </Space>
      </DemoBlock>
    </>
  );
};

      ↓ ↓ ↓ ↓ ↓ ↓

import _intl from "intl";
import React from "react";
import { Button, Space } from "components";
import { DemoBlock } from "demos";
import { SearchOutline } from "icons";
export default () => {
  // 注释a
  const a = _intl.t("intl1");

  const b = \`\${_intl.t("intl2")}\`; // 注释b

  const abc = \`\${a}\${b}\${_intl.t("intl1")}\`;
  return (
    <>
      <DemoBlock title={_intl.t("intl1")}>
        <Button>
          <Space>
            <SearchOutline />
            <span>搜索</span>
          </Space>
        </Button>
      </DemoBlock>

      <DemoBlock title={_intl.t("intl2")}>
        <Space wrap>
          <Button disabled>Disabled</Button>
          <Button disabled color="primary">
            Disabled
          </Button>
        </Space>
      </DemoBlock>

      <DemoBlock title={\`\${loadingTitle}\${_intl.t("intl3")}\`}>
        <Space wrap>
          <Button
            loading
            color={_intl.t("intl4")}
            loadingText={_intl.t("intl5")}
          >
            Loading
          </Button>
          <Button loading>Loading</Button>
        </Space>
      </DemoBlock>
    </>
  );
};


`;
